sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/PartitionKeyRange.java

```diff
@@ -76,20 +78,28 @@ public String getMinInclusive() {
         return super.getString("minInclusive");
     }
 
-    public void setMinInclusive(String minInclusive) {
+    private void setMinInclusive(String minInclusive) {
         BridgeInternal.setProperty(this, "minInclusive", minInclusive);
     }
 
     public String getMaxExclusive() {
         return super.getString("maxExclusive");
     }
 
-    public void setMaxExclusive(String maxExclusive) {
+    private void setMaxExclusive(String maxExclusive) {
         BridgeInternal.setProperty(this, "maxExclusive", maxExclusive);
     }
 
     public Range<String> toRange() {
-        return new Range<String>(this.getMinInclusive(), this.getMaxExclusive(), true, false);
+        if (this.range == null) {
+            synchronized (this) {
+                if (range != null) {
```

Comment:
Should'n it be range  == null

Reply:
this was on earlier commit. already fixed.